src/main/java/com/echainika/app/service/AdminService.java

```diff
@@ -42,8 +46,17 @@ public String bulkDelete() {
     }
 
 
-    public String getAllCandidates(Integer numberOfEntries, Integer pageNumber) {
-        return String.format("Giving %d entries in page number %d", numberOfEntries, pageNumber);
+    public AllCandidatesResponse getAllCandidates(Integer numberOfEntries, Integer pageNumber) {
+        Pageable pageable = PageRequest.of(pageNumber, numberOfEntries);
+        Page<CandidateEntity> candidates = candidateRepository.findAll(pageable);
+        int startIndex = pageNumber*numberOfEntries + 1;
+
+        return AllCandidatesResponse.builder()
+                .candidates(candidates.stream().map(CandidateMapperUtil::candidateMapper).toList())
+                .totalCandidates(candidates.getTotalElements())
+                .startIndex(startIndex)

```

Comment:
to discuss: better to keep pagenumber, and numOfEntries at UI too, why start /end index required